pkg/agent/openflow/feature_templates.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openflow
+
+import (
+	binding "antrea.io/antrea/pkg/ovs/openflow"
+	"antrea.io/antrea/pkg/util/runtime"
+)
+
+func (c *featurePodConnectivity) getTemplate(protocol ofProtocol) *pipelineTemplate {
+	var template *pipelineTemplate
+	if protocol == ofProtocolIP {
+		template = &pipelineTemplate{
+			stageTables: map[binding.StageID][]tableRequest{
+				binding.ClassifierStage: {
+					tableRequest{ClassifierTable, 0x7f},
+				},
+				binding.ValidationStage: {
+					tableRequest{SpoofGuardTable, 0x7f},
+				},
+				binding.ConntrackStateStage: {
+					tableRequest{ConntrackTable, 0x7f},
+					tableRequest{ConntrackStateTable, 0x6f},
+				},
+				binding.RoutingStage: {
+					tableRequest{L3ForwardingTable, 0x7f},
+					tableRequest{L3DecTTLTable, 0x7d},
+				},
+				binding.SwitchingStage: {
+					tableRequest{L2ForwardingCalcTable, 0x7f},
+				},
+				binding.ConntrackStage: {
+					tableRequest{ConntrackCommitTable, 0x7f},
+				},
+				binding.OutputStage: {
+					tableRequest{L2ForwardingOutTable, 0x7f},
+				},
+			},
+		}
+		for _, ipProtocol := range c.ipProtocols {
+			if ipProtocol == binding.ProtocolIPv6 {
+				template.stageTables[binding.ValidationStage] = append(template.stageTables[binding.ValidationStage], tableRequest{IPv6Table, 0x7e})
+				break
+			}
+		}
+	} else if protocol == ofProtocolARP {
```

Comment:
I think here you can use `switch` to make the whole structure more clear.

Reply:
Thanks, updated in #3058